tests/test_relative_free_energy.py

```diff
@@ -183,6 +180,82 @@ def test_steps_per_frames():
         np.testing.assert_array_equal(frame[-1], test_frame[-1])
 
 
+def test_imaging_frames():
+    """Verify that imaging frames places ligand at center and all coordinates are close to being within the box."""
+    mol_a, mol_b, core = get_hif2a_ligand_pair_single_topology()
+    forcefield = Forcefield.load_from_file(DEFAULT_FF)
+    seed = 2022
+    frames = 1
+    steps_per_frame = 100
+    equil_steps = 1
+    windows = 2
+    res, _ = run_solvent(
+        mol_a,
+        mol_b,
+        core,
+        forcefield,
+        None,
+        frames,
+        seed,
+        n_eq_steps=equil_steps,
+        steps_per_frame=steps_per_frame,
+        n_windows=windows,
+        image_traj=False,
+    )
+    keep_idxs = [0, len(res.initial_states) - 1]
+    assert len(keep_idxs) == len(res.frames)
+
+    # A buffer, as imaging doesn't ensure everything is perfectly in the box
+    padding = 0.3
+    for i, (frames, boxes) in enumerate(zip(res.frames, res.boxes)):
+        initial_state = res.initial_states[keep_idxs[i]]
+        box_center = compute_box_center(boxes[0])
+        box_extents = np.max(boxes, axis=(0, 1))
+
+        # Verify that coordinates are either outside of the box or below zero
+        assert np.any(np.max(frames, axis=(0, 1)) > box_extents + padding) or np.any(
+            np.min(frames, axis=(0, 1)) < -padding
+        )
+        # Ligand won't be near center of box
+        assert not np.allclose(np.mean(frames[0][initial_state.ligand_idxs], axis=0), box_center)
+
+        imaged = image_frames(initial_state, frames, boxes)
+
+        # Verify that after imaged, coordinates are within padding of the box extents
+        assert np.all(np.max(imaged, axis=(0, 1)) <= box_extents + padding) and np.all(
+            np.min(imaged, axis=(0, 1)) >= -padding
+        )
+        # Verify that ligand was centered in the box
+        np.testing.assert_allclose(np.mean(imaged[0][initial_state.ligand_idxs], axis=0), box_center)
+
+        for precision, rtol, atol in [(np.float64, 1e-8, 1e-8), (np.float32, 7e-3, 5e-4)]:
```

Comment:
Surprised by the large relative tolerance `7e-3` here. Specifically it is the Nonbonded potential that fails here with a lower tolerance.